File: 11-Bit_Manipulation/1318-minimum_flips_to_make_a_OR_b_equal_to_c.py
# ----------------------------------------------------
#  1318. Minimum Flips to Make a OR b Equal to c (19/08/25); JDG - Py Solution
#  Time complexity : O(n)
#  Space complexity: O(1)
#  ----------------------------------------------------

class Solution:
    def minFlips(self, a: int, b: int, c:int) -> int:
        count = 0
        while(a or b or c):
            aBit = a & 1
            bBit = b & 1
            cBit = c & 1

            if((aBit | bBit) != cBit):
                if(cBit == 1):
                    count += 1
                else:
                    count += (aBit + bBit)

            a >>= 1
            b >>= 1
            c >>= 1

        return count

if __name__ == "main":
    a = 2
    b = 6
    c = 5

    solution = Solution()

    print(solution.minFlips(2, 6, 5))

File: 11-Bit_Manipulation/1318-minimum_flips_to_make_a_OR_b_equal_to_c.cpp
/* ----------------------------------------------------
   1318. Minimum Flips to Make a OR b Equal to c (17/08/25); JDG - CPP Solution
   Time complexity : O(n)
   Space complexity: O(1)
   ---------------------------------------------------- */

#include <iostream>

class Solution {
public:
    int minFlips(int a, int b, int c) {
        int count = 0;
        while (a || b || c) {
            int aBit = a & 1;
            int bBit = b & 1;
            int cBit = c & 1;

            if ((aBit | bBit) != cBit) {
                if (cBit == 1) {
                    ++count;
                } else {
                    count += (aBit + bBit);
                }
            }
            a >>= 1;
            b >>= 1;
            c >>= 1;
        }
        return count;
    }
};

int main() {
    int a = 2;
    int b = 6;
    int c = 5;

    Solution solution;

    std::cout << solution.minFlips(a, b, c) << '\n';

    return 0;
}